packages/core/src/components/exam/DNDAnswerContainer.tsx

```diff
@@ -0,0 +1,208 @@
+  DragStartEvent,
+  KeyboardSensor,
+  PointerSensor,
+  UniqueIdentifier,
+  useSensor,
+  useSensors
+} from '@dnd-kit/core'
+import { sortableKeyboardCoordinates } from '@dnd-kit/sortable'
+import React, { useEffect, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { ExamComponentProps } from '../..'
+import { query, queryAll } from '../../dom-utils'
+import { saveAnswer } from '../../store/answers/actions'
+import { AnswersState } from '../../store/answers/reducer'
+import { DNDTitleAndDroppable } from '../shared/DNDTitleAndDroppable'
+import { AllDNDOptions } from './DNDAllOptions'
+import { DNDAnswerOptionDraggable } from './DNDAnswerOptionDraggable'
+
+export type ItemsState = {
+  root: UniqueIdentifier[]
+  [key: UniqueIdentifier]: UniqueIdentifier[]
+}
+
+export const DNDAnswerContainer = ({ element, renderChildNodes }: ExamComponentProps) => {
+  const [activeId, setActiveId] = useState<UniqueIdentifier | null>()
+  const [optionIdsByQuestionId, setOptionIdsByQuestionId] = useState<ItemsState>({} as ItemsState)
+  const [answerOptionsById, setAnswerOptionsById] = useState<Record<UniqueIdentifier, Element>>({})
+  const [displayNumbersById, setDisplayNumbersById] = useState<Record<UniqueIdentifier, string>>({})
+  const dispatch = useDispatch()
+
+  const answers = useSelector((state: { answers: AnswersState }) => state.answers)
+
+  useEffect(() => {
+    const dndAnswers = queryAll(element, 'dnd-answer').filter(e => !!query(e, 'dnd-answer-title'))
+    const dndAnswerOptions = queryAll(element, 'dnd-answer-option')
+
+    const answerOptionIdsByQuestionId = dndAnswers.reduce(
+      (acc, group) => {
+        const questionId = group.getAttribute('question-id')!
+        const answer = answers.answersById[Number(questionId)]?.value
+        return {
+          ...acc,
+          [questionId]: answer ? [Number(answer)] : [],
+          root: acc.root.filter(e => e !== Number(answer))
+        }
+      },
+      { root: dndAnswerOptions.map(e => Number(e.getAttribute('option-id')!)) }
+    )
+
+    const answerOptionsByOptionId = dndAnswerOptions.reduce(
+      (acc, el) => {
+        const optionId = el.getAttribute('option-id')!
+        return { ...acc, [optionId]: el }
+      },
+      {} as Record<UniqueIdentifier, Element>
+    )
+
+    const displayNumbersById = dndAnswers.reduce(
+      (acc, el) => {
+        const questionId = el.getAttribute('question-id')!
+        const displayNumber = el.getAttribute('display-number')!
+        return { ...acc, [questionId]: displayNumber }
+      },
+      {} as Record<UniqueIdentifier, string>
+    )
+
+    setOptionIdsByQuestionId(answerOptionIdsByQuestionId)
+    setAnswerOptionsById(answerOptionsByOptionId)
+    setDisplayNumbersById(displayNumbersById)
+  }, [element])
+
+  const sensors = useSensors(
+    useSensor(PointerSensor),
+    useSensor(KeyboardSensor, {
+      coordinateGetter: sortableKeyboardCoordinates
+    })
+  )
```

Comment:
kannattaisko tää ottaa pois jos se ei toimi kunnolla?

Reply:
varmaan joo, siis nythän se ei toimi ollenkaan

Reply:
toisaalta ton vois saada väkertäen jotenki toimimaan. jos otetaan pois niin ei muisteta että tuollainen out of the box solution on olemassa.